src/os/bluestore/BlueStore.cc

```diff
@@ -13220,7 +13220,7 @@ int BlueStore::_do_alloc_write(
       // do an approximate (fast) estimation for resulting blob size
       // that doesn't take header overhead  into account
       uint64_t result_len = p2roundup(compressed_len, min_alloc_size);
-      if (r == 0 && result_len <= want_len && result_len < wi.blob_length) {
+      if (r == 0 && result_len <= want_len && result_len <= wi.blob_length) {
```

Comment:
There is no sense in keeping compressed data if compressed length == original length, one gets compressed data without any actual saving. And corresponding decompression penalty on reading...

Reply:
So IMO this PR makes no sense...

Reply:
I concur with you,there is less space saving and more compression cost.I just misunderstand compression_mode=force.  It is a good choice to reject the compression like this. Thx!